elements/pf-popover/pf-popover.ts

```diff
@@ -0,0 +1,389 @@
+
+  // todo: alertSeverityScreenReaderText
+  // todo: hideOnOutsideClick
+  // todo: withFocusTrap
+  @property({ reflect: true }) position: Placement = 'top';
+  @property({ reflect: true }) heading?: string;
+  @property({ reflect: true }) body = '';
+  @property({ reflect: true }) footer?: string;
+  @property({ reflect: true }) icon?: string;
+  @property({ reflect: true }) label?: string;
+  @property({ type: Number, reflect: true }) distance?: number = 25;
+  // todo: handle PF4s 'flip' Placement option in flip-behavior if possible
+  // https://www.patternfly.org/v4/components/popover#popover
+  @property({ attribute: 'flip-behavior', converter: StringListConverter }) flipBehavior?: Placement[] = ['top', 'bottom', 'left', 'right', 'top-start', 'top-end', 'bottom-start', 'bottom-end', 'left-start', 'left-end', 'right-start', 'right-end'];
+  @property({ type: Boolean, reflect: true, attribute: 'enable-flip' }) enableFlip?: boolean = true;
+  @property({ type: Number, reflect: true, attribute: 'heading-level' }) headingLevel?: HeadingLevel;
+  @property({ reflect: true, attribute: 'icon-set' }) iconSet?: string;
+  @property({ type: Boolean, reflect: true, attribute: 'hide-close' }) hideClose?: boolean;
+  @property({ reflect: true, attribute: 'alert-severity' }) alertSeverity?: AlertSeverity;
+  @property({ reflect: true, attribute: 'close-label' }) closeButtonLabel = 'Close popover';
+
+  @observed
+  @property({ reflect: true }) trigger?: string;
+
+  @query('#popover') private _popover?: HTMLDialogElement | null;
+  @query('#trigger') private _slottedTrigger?: HTMLElement | null;
+  @query('#arrow') private _arrow?: HTMLElement | null;
+
+  #referenceTrigger?: HTMLElement | null = null;
+
+  #float = new FloatingDOMController(this, {
+    content: () => this._popover,
+    arrow: () => this._arrow,
+    invoker: () => this.#referenceTrigger || this._slottedTrigger
+  });
+
+  #slots = new SlotController(this, { slots: [null, 'icon', 'heading', 'body', 'footer'] });
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('keydown', this._onKeydown);
+    document.addEventListener('click', this._outsideClick);
+  }
+
+  render() {
+    const { alignment, anchor, styles } = this.#float;
+    return html`
+      <div
+        id="container"
+        style="${styleMap(styles)}"
+        class="${classMap({
+          [anchor]: !!anchor,
+          [alignment]: !!alignment,
+        })}"
+      >
+        <slot id="trigger" @keydown=${this._onKeydown} @click=${this.show}></slot>
+        <dialog
+          id="popover"
+          aria-labelledby="heading"
+          aria-describedby="body"
+          aria-label=${ifDefined(this.label)}
+        > 
+          <div id="arrow"></div>
+          <div id="content" part="content">
+            ${this._renderCloseButton()} ${this._renderHeader()} ${this._renderBody()} ${this._renderFooter()}
+          </div>
+      </dialog>
+      </div>
```

Comment:
```suggestion
      <div id="container"
            style="${styleMap(styles)}"
            class="${classMap({
              [anchor]: !!anchor,
              [alignment]: !!alignment,
            })}">
        <slot id="trigger" @keydown=${this._onKeydown} @click=${this.show}></slot>
        <dialog id="popover"
                aria-labelledby="heading"
                aria-describedby="body"
                aria-label=${ifDefined(this.label)}> 
          <div id="arrow"></div>
          <div id="content" part="content">
            ${this._renderCloseButton()}
            ${this._renderHeader()}
            ${this._renderBody()}
            ${this._renderFooter()}
          </div>
        </dialog>
      </div>
```
just formatting nits